beacon-chain/core/helpers/signing_root.go

```diff
@@ -130,10 +134,6 @@ func RetrieveBlockSignatureSet(blk *ethpb.BeaconBlock, pub []byte, signature []b
 	if err != nil {
 		return nil, errors.Wrap(err, "could not convert bytes to public key")
 	}
-	sig, err := bls.SignatureFromBytes(signature)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not convert bytes to signature")
-	}
```

Comment:
Is this intentional? I mean if we use `RetrieveBlockSignatureSet()` in `VerifyBlockSigningRoot()` (where the SignatureFromBytes check was moved) only -- everything is semantically the same. However, this method is also used in `BlockSignatureSet()` -- and with this change, on that code path we seem to drop the SignatureFromBytes check.

Reply:
Yeap it is fine, as eventually we do verify the signature set. So in the event that it is bad, we will reject the whole set anyway.